src/components/radio-group-item/radio-group-item.tsx

```diff
@@ -84,10 +116,12 @@ export class RadioGroupItem {
             "label--outline": appearance === "outline"
           }}
         >
-          {this.icon && this.iconPosition === "start" ? iconEl : null}
+          {this.icon && this.iconPosition === "start" && !this.iconStart ? iconEl : null}
+          {this.iconStart ? iconStartEl : null}
           <slot>{value}</slot>
           <slot name={SLOTS.input} />
-          {this.icon && this.iconPosition === "end" ? iconEl : null}
+          {this.icon && this.iconPosition === "end" && !this.iconEnd ? iconEl : null}
```

Comment:
Can you refactor this to just render `iconEnd` like you have in the accordion? That way its less confusing here.

Reply:
I don't think we can render iconEnd here as we did in accordion, the issue is with the mapping of the icon to `iconEnd` when `iconPosition` is set to `end` and able to mutate either of `iconStart` or `iconEnd`   when `iconPosition` changes but not `icon` value.

Reply:
I think these should be updated into one line then so its clear they both can't be true.

```
{this.iconEnd ? iconEndEl : this.icon && this.iconPosition === "end" ? iconEl : null}
```

Reply:
sorry to misunderstand, I thought you were suggesting rendering `iconStart` instead of both `iconStart`  &  conditional `icon` with `iconPosition`